src/uast.cc

```diff
@@ -200,9 +204,21 @@ void UastIteratorFree(UastIterator *iter) {
   }
 }
 
+UastIterator *UastIteratorNewWithTransformer(const Uast *ctx, void *node,
+                                             TreeOrder order, void*(*transform)(void*)) {
+
+  UastIterator *iter = UastIteratorNewBase(ctx, node, order);
+  iter->pending.push_front(transform(node));
+  iter->nodeTransform = transform;
+  return iter;
+}
+
 void *UastIteratorNext(UastIterator *iter) {
 
   if (iter == nullptr || iter->pending.empty()) {
+    if (iter == nullptr) {
+    } else if (iter->pending.empty()) {
```

Comment:
I feel that smth was supposed to be said here...

Reply:
Debug crap left behind, I'll change it, thanks.